The job **`aarch64-gnu-llvm-19-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16075581515/job/45370018342) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/45370018342) [(plain)](https://github.com/rust-lang/rust/commit/ccbaf66b6e4410a8c201c587e82a7eb5e3b398b9/checks/45370018342/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/const-generics/defaults/wfness/wfness.stderr`
diff of stderr:

12    |
13    = help: the trait `Trait<2>` is not implemented for `()`
14            but trait `Trait<3>` is implemented for it
+ note: required by a bound in `WhereClause`
+   --> $DIR/wfness.rs:8:9
+    |
+ LL | struct WhereClause<const N: u8 = 2>
+    |        ----------- required by a bound in this struct
+ LL | where
+ LL |     (): Trait<N>;
+    |         ^^^^^^^^ required by this bound in `WhereClause`
15 
16 error[E0277]: the trait bound `(): Trait<1>` is not satisfied
17   --> $DIR/wfness.rs:18:13

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/const-generics/defaults/wfness.rs:8:9
+ note: required by a bound in `WhereClause`
+   --> $DIR/wfness.rs:8:9
+    |
+ LL | struct WhereClause<const N: u8 = 2>
+    |        ----------- required by a bound in this struct
+ LL | where
+ LL |     (): Trait<N>;
+    |         ^^^^^^^^ required by this bound in `WhereClause`


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args const-generics/defaults/wfness.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/const-generics/defaults/wfness.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/const-generics/defaults/wfness" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0080]: attempt to compute `u8::MAX + 1_u8`, which would overflow
##[error]  --> /checkout/tests/ui/const-generics/defaults/wfness.rs:1:33
   |
LL | struct Ooopsies<const N: u8 = { u8::MAX + 1 }>;
   |                                 ^^^^^^^^^^^ evaluation of `Ooopsies::{constant#0}` failed here

error[E0277]: the trait bound `(): Trait<2>` is not satisfied
##[error]  --> /checkout/tests/ui/const-generics/defaults/wfness.rs:8:9
   |
LL |     (): Trait<N>;
   |         ^^^^^^^^ the trait `Trait<2>` is not implemented for `()`
   |
   = help: the trait `Trait<2>` is not implemented for `()`
           but trait `Trait<3>` is implemented for it
note: required by a bound in `WhereClause`
  --> /checkout/tests/ui/const-generics/defaults/wfness.rs:8:9
   |
LL | struct WhereClause<const N: u8 = 2>
   |        ----------- required by a bound in this struct
LL | where
LL |     (): Trait<N>;
   |         ^^^^^^^^ required by this bound in `WhereClause`

error[E0277]: the trait bound `(): Trait<1>` is not satisfied
##[error]  --> /checkout/tests/ui/const-generics/defaults/wfness.rs:18:13
   |
LL | fn foo() -> DependentDefaultWfness {
   |             ^^^^^^^^^^^^^^^^^^^^^^ the trait `Trait<1>` is not implemented for `()`
   |
   = help: the trait `Trait<1>` is not implemented for `()`
           but trait `Trait<3>` is implemented for it
note: required by a bound in `WhereClause`
  --> /checkout/tests/ui/const-generics/defaults/wfness.rs:8:9
   |
LL | struct WhereClause<const N: u8 = 2>
   |        ----------- required by a bound in this struct
---
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/type/type-check-defaults/type-check-defaults.stderr`
diff of stderr:

29    |
30 LL | struct Bounds<T:Copy=String>(T);
31    |                 ^^^^ the trait `Copy` is not implemented for `String`
+    |
+ note: required by a bound in `Bounds`
+   --> $DIR/type-check-defaults.rs:11:17
+    |
+ LL | struct Bounds<T:Copy=String>(T);
+    |                 ^^^^ required by this bound in `Bounds`
32 
33 error[E0277]: the trait bound `String: Copy` is not satisfied
34   --> $DIR/type-check-defaults.rs:14:42

35    |
36 LL | struct WhereClause<T=String>(T) where T: Copy;
37    |                                          ^^^^ the trait `Copy` is not implemented for `String`
+    |
+ note: required by a bound in `WhereClause`
+   --> $DIR/type-check-defaults.rs:14:42
+    |
+ LL | struct WhereClause<T=String>(T) where T: Copy;
+    |                                          ^^^^ required by this bound in `WhereClause`
38 
39 error[E0277]: the trait bound `String: Copy` is not satisfied
40   --> $DIR/type-check-defaults.rs:17:20

41    |
42 LL | trait TraitBound<T:Copy=String> {}
43    |                    ^^^^ the trait `Copy` is not implemented for `String`
+    |
+ note: required by a bound in `TraitBound`
+   --> $DIR/type-check-defaults.rs:17:20
+    |
+ LL | trait TraitBound<T:Copy=String> {}
+    |                    ^^^^ required by this bound in `TraitBound`
44 
45 error[E0277]: the trait bound `T: Copy` is not satisfied
46   --> $DIR/type-check-defaults.rs:21:25

70              `&i32` implements `Add`
71              `i32` implements `Add<&i32>`
72              `i32` implements `Add`
+ note: required by a bound in `ProjectionPred`
+   --> $DIR/type-check-defaults.rs:24:66
+    |
+ LL | trait ProjectionPred<T:Iterator = IntoIter<i32>> where T::Item : Add<u8> {}
+    |                                                                  ^^^^^^^ required by this bound in `ProjectionPred`
73 
74 error: aborting due to 7 previous errors
75 

Note: some mismatched output was normalized before being compared
---
+    |
+ note: required by a bound in `Bounds`
+   --> $DIR/type-check-defaults.rs:11:17
+    |
+ LL | struct Bounds<T:Copy=String>(T);
+    |                 ^^^^ required by this bound in `Bounds`
+    |
+ note: required by a bound in `WhereClause`
+   --> $DIR/type-check-defaults.rs:14:42
+    |
+ LL | struct WhereClause<T=String>(T) where T: Copy;
+    |                                          ^^^^ required by this bound in `WhereClause`
+    |
+ note: required by a bound in `TraitBound`
+   --> $DIR/type-check-defaults.rs:17:20
+    |
+ LL | trait TraitBound<T:Copy=String> {}
+    |                    ^^^^ required by this bound in `TraitBound`
+ note: required by a bound in `ProjectionPred`
+   --> $DIR/type-check-defaults.rs:24:66
+    |
+ LL | trait ProjectionPred<T:Iterator = IntoIter<i32>> where T::Item : Add<u8> {}
+    |                                                                  ^^^^^^^ required by this bound in `ProjectionPred`


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args type/type-check-defaults.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/type/type-check-defaults.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/type/type-check-defaults" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0277]: a value of type `i32` cannot be built from an iterator over elements of type `i32`
##[error]  --> /checkout/tests/ui/type/type-check-defaults.rs:6:23
   |
LL | struct WellFormed<Z = Foo<i32, i32>>(Z);
   |                       ^^^^^^^^^^^^^ value of type `i32` cannot be built from `std::iter::Iterator<Item=i32>`
   |
   = help: the trait `FromIterator<i32>` is not implemented for `i32`
note: required by a bound in `Foo`
  --> /checkout/tests/ui/type/type-check-defaults.rs:5:18
   |
LL | struct Foo<T, U: FromIterator<T>>(T, U);
   |                  ^^^^^^^^^^^^^^^ required by this bound in `Foo`

error[E0277]: a value of type `i32` cannot be built from an iterator over elements of type `i32`
##[error]  --> /checkout/tests/ui/type/type-check-defaults.rs:8:38
   |
LL | struct WellFormedNoBounds<Z:?Sized = Foo<i32, i32>>(Z);
   |                                      ^^^^^^^^^^^^^ value of type `i32` cannot be built from `std::iter::Iterator<Item=i32>`
   |
   = help: the trait `FromIterator<i32>` is not implemented for `i32`
note: required by a bound in `Foo`
  --> /checkout/tests/ui/type/type-check-defaults.rs:5:18
   |
LL | struct Foo<T, U: FromIterator<T>>(T, U);
   |                  ^^^^^^^^^^^^^^^ required by this bound in `Foo`

error[E0277]: the trait bound `String: Copy` is not satisfied
##[error]  --> /checkout/tests/ui/type/type-check-defaults.rs:11:17
   |
LL | struct Bounds<T:Copy=String>(T);
   |                 ^^^^ the trait `Copy` is not implemented for `String`
   |
note: required by a bound in `Bounds`
  --> /checkout/tests/ui/type/type-check-defaults.rs:11:17
   |
LL | struct Bounds<T:Copy=String>(T);
   |                 ^^^^ required by this bound in `Bounds`

error[E0277]: the trait bound `String: Copy` is not satisfied
##[error]  --> /checkout/tests/ui/type/type-check-defaults.rs:14:42
   |
LL | struct WhereClause<T=String>(T) where T: Copy;
   |                                          ^^^^ the trait `Copy` is not implemented for `String`
   |
note: required by a bound in `WhereClause`
  --> /checkout/tests/ui/type/type-check-defaults.rs:14:42
   |
LL | struct WhereClause<T=String>(T) where T: Copy;
   |                                          ^^^^ required by this bound in `WhereClause`

error[E0277]: the trait bound `String: Copy` is not satisfied
##[error]  --> /checkout/tests/ui/type/type-check-defaults.rs:17:20
   |
LL | trait TraitBound<T:Copy=String> {}
   |                    ^^^^ the trait `Copy` is not implemented for `String`
   |
note: required by a bound in `TraitBound`
  --> /checkout/tests/ui/type/type-check-defaults.rs:17:20
   |
LL | trait TraitBound<T:Copy=String> {}
   |                    ^^^^ required by this bound in `TraitBound`

error[E0277]: the trait bound `T: Copy` is not satisfied
##[error]  --> /checkout/tests/ui/type/type-check-defaults.rs:21:25
   |
LL | trait Base<T = String>: Super<T> { }
   |                         ^^^^^^^^ the trait `Copy` is not implemented for `T`
   |
note: required by a bound in `Super`
  --> /checkout/tests/ui/type/type-check-defaults.rs:20:16
   |
LL | trait Super<T: Copy> { }
   |                ^^^^ required by this bound in `Super`
help: consider further restricting type parameter `T` with trait `Copy`
   |
LL | trait Base<T = String>: Super<T> where T: std::marker::Copy { }
   |                                  ++++++++++++++++++++++++++

error[E0277]: cannot add `u8` to `i32`
##[error]  --> /checkout/tests/ui/type/type-check-defaults.rs:24:66
   |
LL | trait ProjectionPred<T:Iterator = IntoIter<i32>> where T::Item : Add<u8> {}
   |                                                                  ^^^^^^^ no implementation for `i32 + u8`
   |
   = help: the trait `Add<u8>` is not implemented for `i32`
   = help: the following other types implement trait `Add<Rhs>`:
             `&i32` implements `Add<i32>`
             `&i32` implements `Add`
             `i32` implements `Add<&i32>`
             `i32` implements `Add`
note: required by a bound in `ProjectionPred`
  --> /checkout/tests/ui/type/type-check-defaults.rs:24:66
   |
LL | trait ProjectionPred<T:Iterator = IntoIter<i32>> where T::Item : Add<u8> {}
   |                                                                  ^^^^^^^ required by this bound in `ProjectionPred`

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0277`.
------------------------------------------
```

</details>

--- a/compiler/rustc_hir_analysis/src/check/check.rs
+++ b/compiler/rustc_hir_analysis/src/check/check.rs
@@ -768,15 +768,14 @@ pub(crate) fn check_item_type(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Result<(),
                     check_static_inhabited(tcx, def_id);
                     check_static_linkage(tcx, def_id);
                     res = res.and(wfcheck::check_static_item(tcx, def_id));
-
-                    // Only `Node::Item` and `Node::ForeignItem` still have HIR based
-                    // checks. Returning early here does not miss any checks and
-                    // avoids this query from having a direct dependency edge on the HIR
-                    return res;
                 }
-                DefKind::Const => {}
+                DefKind::Const => res = res.and(wfcheck::check_const_item(tcx, def_id)),
                 _ => unreachable!(),
             }
+            // Only `Node::Item` and `Node::ForeignItem` still have HIR based
+            // checks. Returning early here does not miss any checks and
+            // avoids this query from having a direct dependency edge on the HIR
+            return res;
         }
         DefKind::Enum => {
             tcx.ensure_ok().generics_of(def_id);

--- a/compiler/rustc_hir_analysis/src/check/wfcheck.rs
+++ b/compiler/rustc_hir_analysis/src/check/wfcheck.rs
@@ -290,7 +290,6 @@ pub(super) fn check_item<'tcx>(
             res
         }
         hir::ItemKind::Fn { sig, .. } => check_item_fn(tcx, def_id, sig.decl),
-        hir::ItemKind::Const(_, _, ty, _) => check_const_item(tcx, def_id, ty.span),
         hir::ItemKind::Struct(..) => check_type_defn(tcx, item, false),
         hir::ItemKind::Union(..) => check_type_defn(tcx, item, true),
         hir::ItemKind::Enum(..) => check_type_defn(tcx, item, true),
@@ -1185,7 +1184,8 @@ pub(super) fn check_static_item(
 ) -> Result<(), ErrorGuaranteed> {
     enter_wf_checking_ctxt(tcx, item_id, |wfcx| {
         let ty = tcx.type_of(item_id).instantiate_identity();
-        let item_ty = wfcx.deeply_normalize(DUMMY_SP, Some(WellFormedLoc::Ty(item_id)), ty);
+        let span = tcx.ty_span(item_id);
+        let item_ty = wfcx.deeply_normalize(span, Some(WellFormedLoc::Ty(item_id)), ty);
 
         let is_foreign_item = tcx.is_foreign_item(item_id);
 
@@ -1194,7 +1194,7 @@ pub(super) fn check_static_item(
             !matches!(tail.kind(), ty::Foreign(_))
         };
 
-        wfcx.register_wf_obligation(DUMMY_SP, Some(WellFormedLoc::Ty(item_id)), item_ty.into());
+        wfcx.register_wf_obligation(span, Some(WellFormedLoc::Ty(item_id)), item_ty.into());
         if forbid_unsized {
             let span = tcx.def_span(item_id);
             wfcx.register_bound(
@@ -1216,7 +1216,6 @@ pub(super) fn check_static_item(
             && !tcx.is_thread_local_static(item_id.to_def_id());
 
         if should_check_for_sync {
-            let span = tcx.def_span(item_id);
             wfcx.register_bound(
                 traits::ObligationCause::new(
                     span,
@@ -1232,13 +1231,10 @@ pub(super) fn check_static_item(
     })
 }
 
-fn check_const_item(
-    tcx: TyCtxt<'_>,
-    def_id: LocalDefId,
-    ty_span: Span,
-) -> Result<(), ErrorGuaranteed> {
+pub(crate) fn check_const_item(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Result<(), ErrorGuaranteed> {
     enter_wf_checking_ctxt(tcx, def_id, |wfcx| {
         let ty = tcx.type_of(def_id).instantiate_identity();
+        let ty_span = tcx.ty_span(def_id);
         let ty = wfcx.deeply_normalize(ty_span, Some(WellFormedLoc::Ty(def_id)), ty);
 
         wfcx.register_wf_obligation(ty_span, Some(WellFormedLoc::Ty(def_id)), ty.into());
@@ -1505,7 +1501,7 @@ pub(super) fn check_where_clauses<'tcx>(wfcx: &WfCheckingCtxt<'_, 'tcx>, def_id:
             let cause = traits::ObligationCause::new(
                 sp,
                 wfcx.body_def_id,
-                ObligationCauseCode::WhereClause(def_id.to_def_id(), DUMMY_SP),
+                ObligationCauseCode::WhereClause(def_id.to_def_id(), sp),
             );
             Obligation::new(tcx, cause, wfcx.param_env, pred)
         });

--- a/compiler/rustc_middle/src/hir/mod.rs
+++ b/compiler/rustc_middle/src/hir/mod.rs
@@ -239,8 +239,16 @@ pub fn provide(providers: &mut Providers) {
         let hir_id = tcx.local_def_id_to_hir_id(def_id);
         tcx.hir_opt_ident_span(hir_id)
     };
+    providers.ty_span = |tcx, def_id| {
+        let node = tcx.hir_node_by_def_id(def_id);
+        match node.ty() {
+            Some(ty) => ty.span,
+            None => bug!("{def_id:?} doesn't have a type: {node:#?}"),
+        }
+    };
     providers.fn_arg_idents = |tcx, def_id| {
-        if let Some(body_id) = tcx.hir_node_by_def_id(def_id).body_id() {
+        let node = tcx.hir_node_by_def_id(def_id);
+        if let Some(body_id) = node.body_id() {
             tcx.arena.alloc_from_iter(tcx.hir_body_param_idents(body_id))
         } else if let Node::TraitItem(&TraitItem {
             kind: TraitItemKind::Fn(_, TraitFn::Required(idents)),
@@ -249,7 +257,7 @@ pub fn provide(providers: &mut Providers) {
         | Node::ForeignItem(&ForeignItem {
             kind: ForeignItemKind::Fn(_, idents, _),
             ..
-        }) = tcx.hir_node(tcx.local_def_id_to_hir_id(def_id))
+        }) = node
         {
             idents
         } else {

--- a/compiler/rustc_middle/src/query/mod.rs
+++ b/compiler/rustc_middle/src/query/mod.rs
@@ -1452,6 +1452,13 @@ rustc_queries! {
         feedable
     }
 
+    /// Gets the span for the type of the definition.
+    /// Panics if it is not a definition that has a single type.
+    query ty_span(def_id: LocalDefId) -> Span {
+        desc { |tcx| "looking up span for `{}`'s type", tcx.def_path_str(def_id) }
+        cache_on_disk_if { true }
+    }
+
     query lookup_stability(def_id: DefId) -> Option<attr::Stability> {
         desc { |tcx| "looking up stability of `{}`", tcx.def_path_str(def_id) }
         cache_on_disk_if { def_id.is_local() }

--- a/tests/ui/const-generics/defaults/wfness.stderr
+++ b/tests/ui/const-generics/defaults/wfness.stderr
@@ -12,6 +12,14 @@ LL |     (): Trait<N>;
    |
    = help: the trait `Trait<2>` is not implemented for `()`
            but trait `Trait<3>` is implemented for it
+note: required by a bound in `WhereClause`
+  --> $DIR/wfness.rs:8:9
+   |
+LL | struct WhereClause<const N: u8 = 2>
+   |        ----------- required by a bound in this struct
+LL | where
+LL |     (): Trait<N>;
+   |         ^^^^^^^^ required by this bound in `WhereClause`
 
 error[E0277]: the trait bound `(): Trait<1>` is not satisfied
   --> $DIR/wfness.rs:18:13

--- a/tests/ui/issues/issue-7364.stderr
+++ b/tests/ui/issues/issue-7364.stderr
@@ -1,8 +1,8 @@
 error[E0277]: `RefCell<isize>` cannot be shared between threads safely
-  --> $DIR/issue-7364.rs:4:1
+  --> $DIR/issue-7364.rs:4:15
    |
 LL | static boxed: Box<RefCell<isize>> = Box::new(RefCell::new(0));
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `RefCell<isize>` cannot be shared between threads safely
+   |               ^^^^^^^^^^^^^^^^^^^ `RefCell<isize>` cannot be shared between threads safely
    |
    = help: the trait `Sync` is not implemented for `RefCell<isize>`
    = note: if you want to do aliasing and mutation between multiple threads, use `std::sync::RwLock` instead

--- a/tests/ui/static/issue-24446.stderr
+++ b/tests/ui/static/issue-24446.stderr
@@ -1,8 +1,8 @@
 error[E0277]: `(dyn Fn() -> u32 + 'static)` cannot be shared between threads safely
-  --> $DIR/issue-24446.rs:2:5
+  --> $DIR/issue-24446.rs:2:17
    |
 LL |     static foo: dyn Fn() -> u32 = || -> u32 {
-   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn Fn() -> u32 + 'static)` cannot be shared between threads safely
+   |                 ^^^^^^^^^^^^^^^ `(dyn Fn() -> u32 + 'static)` cannot be shared between threads safely
    |
    = help: the trait `Sync` is not implemented for `(dyn Fn() -> u32 + 'static)`
    = note: shared static variables must have a type that implements `Sync`

--- a/tests/ui/statics/issue-17718-static-sync.stderr
+++ b/tests/ui/statics/issue-17718-static-sync.stderr
@@ -1,8 +1,8 @@
 error[E0277]: `Foo` cannot be shared between threads safely
-  --> $DIR/issue-17718-static-sync.rs:9:1
+  --> $DIR/issue-17718-static-sync.rs:9:13
    |
 LL | static BAR: Foo = Foo;
-   | ^^^^^^^^^^^^^^^ `Foo` cannot be shared between threads safely
+   |             ^^^ `Foo` cannot be shared between threads safely
    |
    = help: the trait `Sync` is not implemented for `Foo`
    = note: shared static variables must have a type that implements `Sync`

--- a/tests/ui/statics/unsizing-wfcheck-issue-127299.stderr
+++ b/tests/ui/statics/unsizing-wfcheck-issue-127299.stderr
@@ -22,10 +22,10 @@ LL |     fn bar() -> i32 where Self: Sized;
    |                     +++++++++++++++++
 
 error[E0277]: `(dyn Qux + 'static)` cannot be shared between threads safely
-  --> $DIR/unsizing-wfcheck-issue-127299.rs:12:1
+  --> $DIR/unsizing-wfcheck-issue-127299.rs:12:13
    |
 LL | static FOO: &Lint = &Lint { desc: "desc" };
-   | ^^^^^^^^^^^^^^^^^ `(dyn Qux + 'static)` cannot be shared between threads safely
+   |             ^^^^^ `(dyn Qux + 'static)` cannot be shared between threads safely
    |
    = help: within `&'static Lint`, the trait `Sync` is not implemented for `(dyn Qux + 'static)`
    = note: required because it appears within the type `&'static (dyn Qux + 'static)`

--- a/tests/ui/type/type-check-defaults.stderr
+++ b/tests/ui/type/type-check-defaults.stderr
@@ -29,18 +29,36 @@ error[E0277]: the trait bound `String: Copy` is not satisfied
    |
 LL | struct Bounds<T:Copy=String>(T);
    |                 ^^^^ the trait `Copy` is not implemented for `String`
+   |
+note: required by a bound in `Bounds`
+  --> $DIR/type-check-defaults.rs:11:17
+   |
+LL | struct Bounds<T:Copy=String>(T);
+   |                 ^^^^ required by this bound in `Bounds`
 
 error[E0277]: the trait bound `String: Copy` is not satisfied
   --> $DIR/type-check-defaults.rs:14:42
    |
 LL | struct WhereClause<T=String>(T) where T: Copy;
    |                                          ^^^^ the trait `Copy` is not implemented for `String`
+   |
+note: required by a bound in `WhereClause`
+  --> $DIR/type-check-defaults.rs:14:42
+   |
+LL | struct WhereClause<T=String>(T) where T: Copy;
+   |                                          ^^^^ required by this bound in `WhereClause`
 
 error[E0277]: the trait bound `String: Copy` is not satisfied
   --> $DIR/type-check-defaults.rs:17:20
    |
 LL | trait TraitBound<T:Copy=String> {}
    |                    ^^^^ the trait `Copy` is not implemented for `String`
+   |
+note: required by a bound in `TraitBound`
+  --> $DIR/type-check-defaults.rs:17:20
+   |
+LL | trait TraitBound<T:Copy=String> {}
+   |                    ^^^^ required by this bound in `TraitBound`
 
 error[E0277]: the trait bound `T: Copy` is not satisfied
   --> $DIR/type-check-defaults.rs:21:25
@@ -70,6 +88,11 @@ LL | trait ProjectionPred<T:Iterator = IntoIter<i32>> where T::Item : Add<u8> {}
              `&i32` implements `Add`
              `i32` implements `Add<&i32>`
              `i32` implements `Add`
+note: required by a bound in `ProjectionPred`
+  --> $DIR/type-check-defaults.rs:24:66
+   |
+LL | trait ProjectionPred<T:Iterator = IntoIter<i32>> where T::Item : Add<u8> {}
+   |                                                                  ^^^^^^^ required by this bound in `ProjectionPred`
 
 error: aborting due to 7 previous errors
 

